starknet-curve/src/ec_point.rs

```diff
@@ -20,6 +25,49 @@ pub struct ProjectivePoint {
     pub infinity: bool,
 }
 
+// Shared function for lambda calculation for both doubling and addition
+fn calculate_lambda(
+    x1: FieldElement,
+    y1: FieldElement,
+    x2: Option<FieldElement>,
+    y2: Option<FieldElement>,
+) -> FieldElement {
+    match (x2, y2) {
+        // Point doubling case
+        (None, None) => {
+            let dividend = FieldElement::THREE * (x1 * x1) + FieldElement::ONE;
+            let divisor_inv = y1.double().invert().expect("Inversion failed in doubling");
+            dividend * divisor_inv
+        }
+        // Point addition case
+        (Some(x2), Some(y2)) => {
+            let numerator = y2 - y1;
+            let denominator_inv = (x2 - x1).invert().expect("Inversion failed in addition");
```

Comment:
Here I would say that there is a danger in doing this here, because previously this operation was done in the logical sequence of the function and therefore at the step where it was done, we were sure that x2 was not equal at x1 (the check was done before). So there was no risk of a null denominator. Whereas with this utility, we can take the risk, due to ignorance of the inside of the function for the user, of using it with an x1 which would be equal to an x2, generating a panic, right?

Reply:
> Here I would say that there is a danger in doing this here, because previously this operation was done in the logical sequence of the function and therefore at the step where it was done, we were sure that x2 was not equal at x1 (the check was done before). So there was no risk of a null denominator. Whereas with this utility, we can take the risk, due to ignorance of the inside of the function for the user, of using it with an x1 which would be equal to an x2, generating a panic, right?

Thank you so much for your review!

Hmm i don't really understand how it can be 0 here

in the function where we use this, there is a check which is 


```rust
if self.x == rhs.x {
    if self.y == rhs.y {
        self.double_assign();
    } else {
        *self = AffinePoint::identity();
    }
    return;
}
```

So basically if i'm not wrong understanting this part, this check ensures that when x1 (which is self.x) and x2 (which is rhs.x) are equal, the function doesn't proceed to the lambda calculation for addition but instead handles it as either a point doubling or a point at infinit


let me know please :)
And thank you!

Reply:
Yes in this case it's true but I mean in general circumstances, as a user I can use this function without checking before that x1 is not equal to x2 leading potentially to a panic. Or maybe we should wrap this into an error?

Reply:
> Yes in this case it's true but I mean in general circumstances, as a user I can use this function without checking before that x1 is not equal to x2 leading potentially to a panic. Or maybe we should wrap this into an error?

hey, thanks for your reply! ok I see what you mean but the function to calculate the lambda is not public so is not necessarily useful to use as a library. I think this should be done in another PR, right?